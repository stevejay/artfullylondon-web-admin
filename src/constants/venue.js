export const VENUE_TYPE_THEATRE = 'Theatre'
export const VENUE_TYPE_ART_GALLERY = 'Art Gallery'
export const VENUE_TYPE_CONCERT_HALL = 'Concert Hall'
export const VENUE_TYPE_EXHIBITION_HALL = 'Exhibition Hall'
export const VENUE_TYPE_PERFORMING_ARTS_CENTRE = 'Performing Arts Centre'
export const VENUE_TYPE_MUSEUM = 'Museum'
export const VENUE_TYPE_CHURCH = 'Church'
export const VENUE_TYPE_CINEMA = 'Cinema'
export const VENUE_TYPE_OTHER = 'Other'

export const ALLOWED_VENUE_TYPES = [
  VENUE_TYPE_THEATRE,
  VENUE_TYPE_ART_GALLERY,
  VENUE_TYPE_CONCERT_HALL,
  VENUE_TYPE_EXHIBITION_HALL,
  VENUE_TYPE_PERFORMING_ARTS_CENTRE,
  VENUE_TYPE_MUSEUM,
  VENUE_TYPE_CHURCH,
  VENUE_TYPE_CINEMA,
  VENUE_TYPE_OTHER
]
