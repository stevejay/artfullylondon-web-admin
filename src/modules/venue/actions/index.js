export const types = {
  GET_VENUE_EVENT_MONITORS: 'venue/GET_VENUE_EVENT_MONITORS',
  GET_VENUE_EVENT_MONITORS_STARTED: 'venue/GET_VENUE_EVENT_MONITORS_STARTED',
  GET_VENUE_EVENT_MONITORS_SUCCEEDED: 'venue/GET_VENUE_EVENT_MONITORS_SUCCEEDED',
  GET_VENUE_EVENT_MONITORS_FAILED: 'venue/GET_VENUE_EVENT_MONITORS_FAILED',
  UPDATE_VENUE_EVENT_MONITOR: 'venue/UPDATE_VENUE_EVENT_MONITOR',
  UPDATE_VENUE_EVENT_MONITOR_SUCCEEDED: 'venue/UPDATE_VENUE_EVENT_MONITOR_SUCCEEDED',
  GET_VENUE_MONITOR: 'venue/GET_VENUE_MONITOR',
  GET_VENUE_MONITOR_STARTED: 'venue/GET_VENUE_MONITOR_STARTED',
  GET_VENUE_MONITOR_SUCCEEDED: 'venue/GET_VENUE_MONITOR_SUCCEEDED',
  GET_VENUE_MONITOR_FAILED: 'venue/GET_VENUE_MONITOR_FAILED',
  UPDATE_VENUE_MONITOR: 'venue/UPDATE_VENUE_MONITOR',
  UPDATE_VENUE_MONITOR_SUCCEEDED: 'venue/UPDATE_VENUE_MONITOR_SUCCEEDED'
}

export const getVenueEventMonitors = venueId => ({
  type: types.GET_VENUE_EVENT_MONITORS,
  payload: { venueId }
})

export const getVenueEventMonitorsStarted = () => ({
  type: types.GET_VENUE_EVENT_MONITORS_STARTED
})

export const getVenueEventMonitorsSucceeded = json => ({
  type: types.GET_VENUE_EVENT_MONITORS_SUCCEEDED,
  payload: json
})

export const getVenueEventMonitorsFailed = () => ({
  type: types.GET_VENUE_EVENT_MONITORS_FAILED
})

export const updateVenueEventMonitor = values => ({
  type: types.UPDATE_VENUE_EVENT_MONITOR,
  payload: { values },
  meta: { thunk: true }
})

export const updateVenueEventMonitorSucceeded = values => ({
  type: types.UPDATE_VENUE_EVENT_MONITOR_SUCCEEDED,
  payload: values
})

export const getVenueMonitor = venueId => ({
  type: types.GET_VENUE_MONITOR,
  payload: { venueId }
})

export const getVenueMonitorStarted = () => ({
  type: types.GET_VENUE_MONITOR_STARTED
})

export const getVenueMonitorSucceeded = json => ({
  type: types.GET_VENUE_MONITOR_SUCCEEDED,
  payload: json
})

export const getVenueMonitorFailed = () => ({
  type: types.GET_VENUE_MONITOR_FAILED
})

export const updateVenueMonitor = values => ({
  type: types.UPDATE_VENUE_MONITOR,
  payload: { values },
  meta: { thunk: true }
})

export const updateVenueMonitorSucceeded = values => ({
  type: types.UPDATE_VENUE_MONITOR_SUCCEEDED,
  payload: values
})
