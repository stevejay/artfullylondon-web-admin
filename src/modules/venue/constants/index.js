import _ from 'lodash'

import wheelchairAccessType from '_src/entities/types/wheelchair-access-type'
import disabledBathroomType from '_src/entities/types/disabled-bathroom-type'
import hearingFacilitiesType from '_src/entities/types/hearing-facilities-type'
import venueType from '_src/entities/types/venue-type'

export const EDIT_VENUE_FORM_NAME = 'EditVenue'

export const HEARING_FACILITIES_TYPE_DROPDOWN_OPTIONS = [
  {
    value: hearingFacilitiesType.NOT_APPLICABLE,
    label: 'Not applicable'
  },
  {
    value: hearingFacilitiesType.HEARING_LOOPS,
    label: 'Hearing loops in all spaces'
  },
  {
    value: hearingFacilitiesType.PARTIAL_HEARING_LOOPS,
    label: 'Hearing loops in some spaces'
  },
  {
    value: hearingFacilitiesType.NO_HEARING_LOOPS,
    label: 'No hearing loops'
  },
  { value: hearingFacilitiesType.UNKNOWN, label: 'Unknown' }
]

export const DISABLED_BATHROOM_TYPE_DROPDOWN_OPTIONS = [
  {
    value: disabledBathroomType.NOT_APPLICABLE,
    label: 'Not applicable'
  },
  { value: disabledBathroomType.PRESENT, label: 'Present' },
  {
    value: disabledBathroomType.NOT_PRESENT,
    label: 'Not present'
  },
  { value: disabledBathroomType.UNKNOWN, label: 'Unknown' }
]

export const WHEELCHAIR_ACCESS_TYPE_DROPDOWN_OPTIONS = [
  {
    value: wheelchairAccessType.NOT_APPLICABLE,
    label: 'Not applicable'
  },
  {
    value: wheelchairAccessType.FULL_ACCESS,
    label: 'Access to all spaces'
  },
  {
    value: wheelchairAccessType.PARTIAL_ACCESS,
    label: 'Access to some spaces'
  },
  {
    value: wheelchairAccessType.NO_ACCESS,
    label: 'No access'
  },
  { value: wheelchairAccessType.UNKNOWN, label: 'Unknown' }
]

export const VENUE_TYPE_DROPDOWN_OPTIONS = [
  {
    value: venueType.THEATRE,
    label: venueType.THEATRE
  },
  {
    value: venueType.ART_GALLERY,
    label: venueType.ART_GALLERY
  },
  {
    value: venueType.CONCERT_HALL,
    label: venueType.CONCERT_HALL
  },
  {
    value: venueType.EXHIBITION_HALL,
    label: 'Exhibition Space'
  },
  {
    value: venueType.PERFORMING_ARTS_CENTRE,
    label: 'Arts Centre'
  },
  {
    value: venueType.MUSEUM,
    label: venueType.MUSEUM
  },
  {
    value: venueType.CHURCH,
    label: venueType.CHURCH
  },
  {
    value: venueType.CINEMA,
    label: venueType.CINEMA
  },
  {
    value: venueType.OTHER,
    label: venueType.OTHER
  }
]

export const VENUE_CONSTRAINT = {
  name: {
    presence: { disallowEmpty: true },
    length: { maximum: 200 }
  },
  venueType: {
    presence: { disallowEmpty: true },
    inclusion: { within: _.values(venueType) }
  },
  address: {
    presence: { disallowEmpty: true },
    length: { maximum: 400 }
  },
  postcode: {
    presence: { disallowEmpty: true },
    length: { maximum: 9 },
    format: { pattern: /^[A-Z]{1,2}[0-9]{1,2}[A-Z]?\s+[0-9][A-Z]{2}$/ }
  },
  email: {
    length: { maximum: 100 },
    format: { pattern: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+.[A-Z]{2,4}$|^$/i }
  },
  telephone: {
    length: { maximum: 20 },
    format: { pattern: /^\d[\d\s-]{6,18}\d$|^$/ }
  },
  wheelchairAccessType: {
    presence: { disallowEmpty: true },
    inclusion: { within: _.values(wheelchairAccessType) }
  },
  disabledBathroomType: {
    presence: { disallowEmpty: true },
    inclusion: { within: _.values(disabledBathroomType) }
  },
  hearingFacilitiesType: {
    presence: { disallowEmpty: true },
    inclusion: { within: _.values(hearingFacilitiesType) }
  },
  images: {
    length: { maximum: 10, tooLong: 'has too many elements' }
  },
  weSay: {
    length: { maximum: 300 }
  },
  notes: {
    length: { maximum: 400 }
  }
}

export const VENUE_NORMALISER = {
  name: {
    trim: true
  },
  address: {
    trim: true
  },
  postcode: {
    trim: true
  },
  email: {
    trim: true
  },
  telephone: {
    trim: true
  },
  weSay: {
    trim: true
  },
  notes: {
    trim: true
  }
}
