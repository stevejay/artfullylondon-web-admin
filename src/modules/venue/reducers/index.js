import { combineReducers } from 'redux'

import * as reduxLib from '_src/lib/redux'

import * as venueMonitor from './venue-monitor'
import * as venueEventMonitors from './venue-event-monitors'

export const moduleName = 'venue'

export const reducer = combineReducers({
  [venueMonitor.moduleName]: venueMonitor.reducer,
  [venueEventMonitors.moduleName]: venueEventMonitors.reducer
})

export const selectors = reduxLib.mapSelectors(
  reduxLib.combineSelectors(
    reduxLib.mapSelectors(venueMonitor.selectors, venueMonitor.moduleName),
    reduxLib.mapSelectors(
      venueEventMonitors.selectors,
      venueEventMonitors.moduleName
    )
  ),
  moduleName
)
