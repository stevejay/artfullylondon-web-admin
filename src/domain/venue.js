import _ from 'lodash'

import * as venueLib from '_src/lib/venue'
import * as timeLib from '_src/lib/time'
import { Entity } from '_src/domain/entity'
import entityType from '_src/domain/types/entity-type'
import linkType from '_src/domain/types/link-type'

const NEWLINES_REGEX = /\n/g

export class FullVenue extends Entity {
  get entityType () {
    return entityType.VENUE
  }

  getInfoBarLabel () {
    return this.venueType
  }

  get url () {
    throw new Error('url accessed')
  }

  get editUrl () {
    return `/venue/edit/${this.id}`
  }

  get pin () {
    return {
      lat: this.latitude,
      lng: this.longitude
    }
  }

  getHomepageUrl () {
    const homepage = this.getLinkByType(linkType.HOMEPAGE)
    return homepage ? homepage.url : null
  }

  createTimesDetailsOn (dateStr) {
    return timeLib.getTimesDetails(this, entityType.VENUE, dateStr)
  }

  shallowClone (newProps) {
    return new FullVenue({ ...this, ...newProps })
  }
}

export class SummaryVenue {
  constructor (entity) {
    _.extend(this, entity)
  }

  getPostcodeDistrict () {
    return venueLib.getPostcodeDistrict(this.postcode)
  }

  get entityTypeLabel () {
    return 'Venue'
  }

  get key () {
    return this.id
  }

  get url () {
    return `/venue/${this.id}`
  }

  get pin () {
    return {
      lat: this.latitude,
      lng: this.longitude
    }
  }

  get hasImage () {
    return !!this.image
  }

  createFullAddress () {
    return (
      this.address.replace(NEWLINES_REGEX, ', ') +
      (this.postcode ? ', ' + this.postcode : '')
    )
  }

  shallowClone (newProps) {
    return new SummaryVenue({ ...this, ...newProps })
  }
}
