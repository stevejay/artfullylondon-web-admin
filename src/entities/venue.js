import { ENTITY_TYPE_VENUE } from '_src/constants/entity'
import { LinkCollection } from '_src/entities/link-collection'
import * as sharedVenue from '_src/lib/venue'
import * as sharedEntity from '_src/lib/entity'
import { LINK_TYPE_HOMEPAGE } from '_src/constants/link'
import * as time from '_src/lib/time'
import * as image from '_src/lib/image'

export class SummaryVenue {
  constructor (entity) {
    if (!entity) {
      throw new Error('null or undefined entity')
    }

    this.entity = entity
    this.postcodeDistrict = sharedVenue.getPostcodeDistrict(entity.postcode)
  }

  get entityType () {
    return ENTITY_TYPE_VENUE
  }

  get entityTypeLabel () {
    return 'Venue'
  }

  get status () {
    return this.entity.status
  }

  get id () {
    return this.entity.id
  }

  get key () {
    return this.id
  }

  get name () {
    return this.entity.name
  }

  get venueType () {
    return this.entity.venueType
  }

  get postcode () {
    return this.entity.postcode
  }

  get url () {
    return sharedEntity.createEntityUrl(this.entityType, this.id)
  }

  get editUrl () {
    return sharedEntity.createEntityEditUrl(this.entityType, this.id)
  }

  get latitude () {
    return this.entity.latitude
  }

  get longitude () {
    return this.entity.longitude
  }

  get pin () {
    return {
      lat: this.latitude,
      lng: this.longitude
    }
  }

  get image () {
    return this.entity.image
  }

  get imageCopyright () {
    return this.entity.imageCopyright
  }

  get imageRatio () {
    return this.entity.imageRatio
  }

  get hasImage () {
    return !!this.image
  }

  get cardImageLoaded () {
    return !!this.entity.cardImageLoaded
  }

  get isFullEntity () {
    return false
  }

  isBeingWatched (watches) {
    return !!watches[this.id]
  }

  createWatchLabel () {
    return this.name
  }

  createWatchChangeInstruction (isBeingWatched) {
    const prefix = isBeingWatched ? 'Unwatch' : 'Watch'
    return prefix + ' this venue'
  }

  createFullAddress () {
    return sharedVenue.formatAddressForDisplay(
      this.entity.address,
      this.entity.postcode
    )
  }

  createVenuesMapIconUrl (isSelected) {
    return image.createVenueTypePngIconUrl(this.venueType, isSelected)
  }

  shallowClone (newProps) {
    const clonedEntity = Object.assign({}, this.entity, newProps || {})
    return new SummaryVenue(clonedEntity)
  }
}

export class FullVenue extends SummaryVenue {
  constructor (entity) {
    super(entity)
    this.links = new LinkCollection(entity.links)
  }

  createTimesDescriptionForDate (dateStr, timeStr, namedClosuresLookup) {
    return time.formatTimesStringForGivenDate(
      this.entity,
      dateStr,
      timeStr,
      namedClosuresLookup
    )
  }

  createTimesDetailsOn (dateStr) {
    return time.getTimesDetails(this.entity, this.entityType, dateStr)
  }

  createInfoBarLabel () {
    return this.venueType
  }

  createFormattedDescription () {
    return sharedEntity.processDescription(
      this.entity.description,
      this.entity.descriptionCredit
    )
  }

  get images () {
    return this.entity.images
  }

  get description () {
    return this.entity.description
  }

  get descriptionCredit () {
    return this.entity.descriptionCredit
  }

  get weSay () {
    return this.entity.weSay
  }

  get email () {
    return this.entity.email
  }

  get telephone () {
    return this.entity.telephone
  }

  get wheelchairAccessType () {
    return this.entity.wheelchairAccessType
  }

  get disabledBathroomType () {
    return this.entity.disabledBathroomType
  }

  get hearingFacilitiesType () {
    return this.entity.hearingFacilitiesType
  }

  get hasPermanentCollection () {
    return this.entity.hasPermanentCollection
  }

  get isFullEntity () {
    return true
  }

  shallowClone (newProps) {
    const clonedEntity = Object.assign({}, this.entity, newProps || {})
    return new FullVenue(clonedEntity)
  }

  getLinkByType (linkType) {
    return this.links.getLinkByType(linkType)
  }

  getHomepageUrl () {
    const homepage = this.getLinkByType(LINK_TYPE_HOMEPAGE)
    return homepage ? homepage.url : null
  }
}
