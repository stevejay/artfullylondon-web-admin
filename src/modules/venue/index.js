import VenueEditOrCreate from './components/venue-edit-or-create'
import VenueDetail from './components/venue-detail'
import { moduleName, reducer, selectors } from './reducers'
import sagas from './sagas'

export { VenueEditOrCreate, VenueDetail, moduleName, reducer, selectors, sagas }
