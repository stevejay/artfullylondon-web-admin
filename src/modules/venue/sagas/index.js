import { call, put, takeEvery, takeLatest } from 'redux-saga/effects'
import { startSubmit, stopSubmit } from 'redux-form'
import log from 'loglevel'

import { get, put as httpPut } from '_src/lib/fetch'
import * as sagaLib from '_src/lib/saga'
import * as validationLib from '_src/lib/validation'
import { getAuthTokenForCurrentUser } from '_src/modules/user'
import * as venueConstants from '_src/modules/venue/constants'
import * as venueActions from '_src/modules/venue/actions'

export function * getVenueEventMonitors (action) {
  try {
    const { venueId } = action.payload
    yield put(venueActions.getVenueEventMonitorsStarted())

    const token = yield call(getAuthTokenForCurrentUser)
    const url = `${process.env.WEBSITE_API_HOST_URL}/monitor-service/monitor/venue/${venueId}/event/`
    const json = yield call(get, url, token)

    json.items.forEach(element => {
      element.key = element.externalEventId
    })

    yield put(venueActions.getVenueEventMonitorsSucceeded(json.items))
  } catch (err) {
    yield call(log.error, err)
    yield put(venueActions.getVenueEventMonitorsFailed)
  }
}

// export function * getVenueEventMonitor (action) {
//   try {
//     const { venueId, externalEventId } = action.payload
//     yield put(venueActions.getVenueEventMonitorStarted())

//     const token = yield call(getAuthTokenForCurrentUser)
//     const url = `${process.env.WEBSITE_API_HOST_URL}/monitor-service/monitor/venue/${venueId}/event/${encodeURIComponent(externalEventId)}`
//     const json = yield call(get, url, token)

//     yield put(venueActions.getVenueEventMonitorSucceeded(json))
//   } catch (err) {
//     yield call(log.error, err)
//     yield put(venueActions.getVenueEventMonitorFailed())
//   }
// }

export function * updateVenueEventMonitor ({ payload: { values }, meta }) {
  try {
    yield put(startSubmit(venueConstants.UPDATE_MONITOR_FORM_NAME))

    yield call(
      validationLib.validate,
      values,
      venueConstants.MONITOR_CONSTRAINT
    )

    const token = yield call(getAuthTokenForCurrentUser)
    const url = `${process.env.WEBSITE_API_HOST_URL}/monitor-service/monitor/venue/${values.venueId}/event/${encodeURIComponent(values.externalEventId)}`

    const body = {
      hasChanged: values.hasChanged,
      isIgnored: values.isIgnored
    }

    yield call(httpPut, url, body, token)
    yield put(venueActions.updateVenueEventMonitorSucceeded(values))
    yield put(stopSubmit(venueConstants.UPDATE_MONITOR_FORM_NAME))
    yield put(sagaLib.returnAsPromise(null, meta))
  } catch (err) {
    yield call(log.error, err)

    yield call(
      sagaLib.submitErrorHandler,
      err,
      venueConstants.UPDATE_MONITOR_FORM_NAME
    )
  }
}

export function * getVenueMonitor (action) {
  try {
    const { venueId } = action.payload
    yield put(venueActions.getVenueMonitorStarted())

    const token = yield call(getAuthTokenForCurrentUser)
    const url = `${process.env.WEBSITE_API_HOST_URL}/monitor-service/monitor/venue/${venueId}`
    const json = yield call(get, url, token)

    const entity = json.entity
    entity.key = entity.venueId
    yield put(venueActions.getVenueMonitorSucceeded(entity))
  } catch (err) {
    if (err.statusCode !== 404) {
      yield call(log.error, err)
    }

    yield put(venueActions.getVenueMonitorFailed())
  }
}

export function * updateVenueMonitor ({ payload: { values }, meta }) {
  try {
    yield put(startSubmit(venueConstants.UPDATE_MONITOR_FORM_NAME))

    yield call(
      validationLib.validate,
      values,
      venueConstants.MONITOR_CONSTRAINT
    )

    const token = yield call(getAuthTokenForCurrentUser)
    const url = `${process.env.WEBSITE_API_HOST_URL}/monitor-service/monitor/venue/${values.venueId}`

    const body = {
      hasChanged: values.hasChanged,
      isIgnored: values.isIgnored
    }

    yield call(httpPut, url, body, token)
    yield put(venueActions.updateVenueMonitorSucceeded(values))
    yield put(stopSubmit(venueConstants.UPDATE_MONITOR_FORM_NAME))
    yield put(sagaLib.returnAsPromise(null, meta))
  } catch (err) {
    yield call(log.error, err)

    yield call(
      sagaLib.submitErrorHandler,
      err,
      venueConstants.UPDATE_MONITOR_FORM_NAME
    )
  }
}

export default [
  takeLatest(
    venueActions.types.GET_VENUE_EVENT_MONITORS,
    getVenueEventMonitors
  ),
  // takeLatest(venueActions.types.GET_VENUE_EVENT_MONITOR, getVenueEventMonitor),
  takeEvery(
    venueActions.types.UPDATE_VENUE_EVENT_MONITOR,
    updateVenueEventMonitor
  ),
  takeLatest(venueActions.types.GET_VENUE_MONITOR, getVenueMonitor),
  takeEvery(venueActions.types.UPDATE_VENUE_MONITOR, updateVenueMonitor)
]
