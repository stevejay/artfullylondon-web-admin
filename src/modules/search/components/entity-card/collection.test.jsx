import React from 'react'
import _ from 'lodash'

import EntityCardCollection from './collection'
import { SummaryVenue } from '_src/domain/venue'

it('should render correctly when get is in progress', () => {
  const wrapper = shallow(
    <EntityCardCollection
      entities={[new SummaryVenue({ id: 'some-venue-id' })]}
      getInProgress
      dateStr='2017/01/18'
      cardContentFactory={() => 'div'}
      onMounted={_.noop}
    />
  )

  expect(wrapper).toMatchSnapshot()
})

it('should render correctly when there are entities', () => {
  const wrapper = shallow(
    <EntityCardCollection
      entities={[new SummaryVenue({ id: 'some-venue-id' })]}
      getInProgress={false}
      dateStr='2017/01/18'
      cardContentFactory={() => 'div'}
    />
  )

  expect(wrapper).toMatchSnapshot()
})

it('should render correctly when there are no entities', () => {
  const wrapper = shallow(
    <EntityCardCollection
      entities={null}
      getInProgress={false}
      dateStr='2017/01/18'
      cardContentFactory={() => 'div'}
    />
  )

  expect(wrapper).toMatchSnapshot()
})
