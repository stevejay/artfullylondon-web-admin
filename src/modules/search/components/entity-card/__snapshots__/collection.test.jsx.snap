// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly when get is in progress 1`] = `
<div
  className="src-modules-search-components-entity-card-collection__cards-container"
>
  <EntityCard
    cardContentFactory={[Function]}
    dateStr="2017/01/18"
    entity={
      SummaryVenue {
        "entity": Object {
          "id": "some-venue-id",
        },
        "postcodeDistrict": null,
      }
    }
    key="some-venue-id"
  />
</div>
`;

exports[`should render correctly when there are entities 1`] = `
<div
  className="src-modules-search-components-entity-card-collection__cards-container"
>
  <EntityCard
    cardContentFactory={[Function]}
    dateStr="2017/01/18"
    entity={
      SummaryVenue {
        "entity": Object {
          "id": "some-venue-id",
        },
        "postcodeDistrict": null,
      }
    }
    key="some-venue-id"
  />
</div>
`;

exports[`should render correctly when there are no entities 1`] = `
<div
  className="src-modules-search-components-entity-card-collection__cards-container"
/>
`;
